homeassistant/components/esphome/domain_data.py

```diff
@@ -70,8 +36,7 @@ def get_entry_data(self, entry: ConfigEntry) -> RuntimeEntryData:
 
     def set_entry_data(self, entry: ConfigEntry, entry_data: RuntimeEntryData) -> None:
         """Set the runtime entry data associated with this config entry."""
-        if entry.entry_id in self._entry_datas:
-            raise ValueError("Entry data for this entry is already set")
+        assert entry.entry_id not in self._entry_datas, "Entry data already set!"
```

Comment:
This should be unreachable but I left it as an assert just in case we do have a bug (seems unlikely now that we have coverage for load/unload)